drv/cosmo-hf/src/hf.rs

```diff
@@ -161,41 +163,29 @@ impl ServerImpl {
     /// Ensures that the persistent data is consistent between the virtual devs
     fn ensure_persistent_data_is_redundant(&mut self) {
         // Load the current state of persistent data from flash
-        let prev_slot = self.dev;
-
-        self.dev = !prev_slot;
-        let (a_data, a_next) = self.persistent_data_scan();
-
-        self.dev = prev_slot;
-        let (b_data, b_next) = self.persistent_data_scan();
-
-        match (a_data, b_data) {
-            (Some(a), Some(b)) => {
-                match a.cmp(&b) {
-                    core::cmp::Ordering::Less => {
-                        self.dev = !prev_slot;
-                        self.write_raw_persistent_data_to_addr(a_next, &b);
-                        self.dev = prev_slot;
-                    }
-                    core::cmp::Ordering::Greater => {
-                        self.write_raw_persistent_data_to_addr(b_next, &a)
-                    }
-                    core::cmp::Ordering::Equal => {
-                        // Redundant data is consistent
-                        // TODO: should we have a special case if they don't agree?
-                    }
-                }
-            }
-            (Some(a), None) => {
-                self.write_raw_persistent_data_to_addr(b_next, &a)
-            }
-            (None, Some(b)) => {
-                self.dev = !prev_slot;
-                self.write_raw_persistent_data_to_addr(a_next, &b);
-                self.dev = prev_slot;
+        let (data0, next0) = self.persistent_data_scan(HfDevSelect::Flash0);
+        let (data1, next1) = self.persistent_data_scan(HfDevSelect::Flash1);
```

Comment:
as an aside, i like that we have gone from `a`/`b` naming here to `0`/`1`, because it matches the `HfDevSelect` terminology! :)